sympy/functions/combinatorial/numbers.py

```diff
@@ -1861,10 +1861,12 @@ def stirling(n, k, d=None, kind=2, signed=False):
 def _nT(n, k):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if n == k or k == 1:
+    if n == k:
         return 1
     if k > n or k == 0:
         return 0
+    if k == 1:
```

Comment:
What values of `n` and `k` will require the 1 to be returned in two different places? i.e. what motivated this change?